contracts/debt/IncurDebtV1.sol

```diff
@@ -0,0 +1,458 @@
+        totalOutstandingGlobalDebt += _ohmAmount;
+
+        ITreasury(treasury).incurDebt(_ohmAmount, OHM);
+        IERC20(OHM).safeTransfer(msg.sender, _ohmAmount);
+
+        emit Borrowed(msg.sender, _ohmAmount, borrower.debt, totalOutstandingGlobalDebt);
+    }
+
+    /**
+     * @notice withdraws gOHM/sOHM  to _to address
+     * - msg.sender must be a borrower
+     * - _amount (in OHM) must be less than or equal to depositedOhm - debt
+     * @param _amount amount of gOHM/sOHM to withdraw
+     * @param _to address to send _amount
+     * @param _token token to send _amount
+     */
+    function withdraw(
+        uint256 _amount,
+        address _to,
+        address _token
+    ) external override isBorrower(msg.sender) {
+        if (_amount == 0) revert IncurDebtV1_InvaildNumber(_amount);
+        Borrower storage borrower = borrowers[msg.sender];
+
+        updateCollateralInSOHM(msg.sender);
+
+        if (_token == gOHM) {
+            uint256 _sAmount = IgOHM(gOHM).balanceFrom(_amount);
+
+            if (_sAmount > getAvailableToBorrow()) revert IncurDebtV1_AmountAboveBorrowerBalance(_sAmount);
+
+            borrower.collateralInSOHM -= uint128(_sAmount);
+            borrower.collateralInGOHM -= uint128(_amount);
+
+            IERC20(sOHM).approve(staking, _sAmount);
+            IStaking(staking).wrap(_to, _sAmount);
+        } else if (_token == sOHM) {
+            if (_amount > getAvailableToBorrow()) revert IncurDebtV1_AmountAboveBorrowerBalance(_amount);
+
+            borrower.collateralInSOHM -= uint128(_amount);
+            borrower.collateralInGOHM -= uint128(IgOHM(gOHM).balanceTo(_amount));
+
+            IERC20(sOHM).safeTransfer(_to, _amount);
+        }
+
+        emit Withdrawal(msg.sender, _amount, _token, _to, borrower.collateralInSOHM);
+    }
+
+    /**
+     * @notice repay debt with collateral
+     * - msg.sender must be a borrower
+     * - borrower must have outstanding debt
+     */
+    function repayDebtWithCollateral() external override {
+        Borrower storage borrower = borrowers[msg.sender];
+        (uint256 depositedCollateralAfterRepay, uint256 paidDebt) = _repay(msg.sender);
+
+        borrower.debt = 0;
+        borrower.collateralInSOHM = uint128(depositedCollateralAfterRepay);
+
+        borrower.collateralInGOHM = uint128(IgOHM(gOHM).balanceTo(depositedCollateralAfterRepay));
+        emit DebtPaidWithCollateral(
+            msg.sender,
+            paidDebt,
+            depositedCollateralAfterRepay,
+            borrower.debt,
+            totalOutstandingGlobalDebt
+        );
+    }
+
+    /**
+     * @notice repay debt with collateral and withdraw the accrued earnings to sOHM/gOHM
+     * - msg.sender must be a borrower
+     * - borrower must have outstanding debt
+     * @param _tokenToReceiveExcess amount of OHM to borrow
+     */
+    function repayDebtWithCollateralAndWithdrawTheRest(address _tokenToReceiveExcess) external override {
```

Comment:
make the arg into a bool. dont let user pass any address.

Reply:
Fixed.